datastore/jdbc-utils/src/main/groovy/org/orbisgis/datastore/jdbcutils/JDBCDataStoreUtils.groovy

```diff
@@ -2462,4 +2436,97 @@ static int[] withBatch(JDBCDataStore ds, int batchSize, String sql,
             @ClosureParams(value=SimpleType.class, options="groovy.sql.BatchingPreparedStatementWrapper") Closure closure)
         throws SQLException {
     getSql(ds).withBatch(batchSize, sql, closure)
+}
+
+/**
+ * Export the given table from the given datastore into a file with the encoding.
+ *
+ * @param connection DataStore containing the table.
+ * @param tableName  Name of the table to save.
+ * @param filePath   Path of the destination file.
+ * @param encoding   Encoding of the file. Set to null by default.
+ * @param deleteFile True to delete the file if exists. Set to true by default.
+ * @throws java.sql.SQLException Exception throw on database error.
+ */
+static void export(JDBCDataStore ds, String tableName, String filePath,
+                         String encoding = null, boolean deleteFile = true)
+        throws SQLException {
+    def type = ds.properties.dataStoreFactory.DBTYPE.sample
+    if(type == "h2gis" || type == "h2" || type == "postgresql" || type == "postgis") {
```

Comment:
Export must be able to work with geotools datastore file.
Here you must use datastorefinder and if there is gt driver -> do the export with if not try with H2GIS utilities according the db type